### PR TITLE
[chore] add port check for dev command

### PR DESCRIPTION
**Pull request by Zerotask**
Close https://github.com/sveltejs/kit/issues/2789

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [ ] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpx changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

### CODE REVIEW

**Comment by bluwy** [comment]
I don't think it's correct to check the ports in dev mode. There is a possibility someone set `kit.vite.server.fs.strictPort: false`, which would search a different port, e.g. `3001` and run that. I think the better way to solve this is to do the checks in Vite instead.

**Comment by Zerotask** [comment]
The current solution (as mentioned in the issue) is not good and not beginner friendly. This should be changed. Either this way or with another way.
Checking ports is pretty straightforward and it's the same function which is used for the preview command. Maybe it could be wrapped around an if statement which checks this property (if it has access to it).

**Comment by bluwy** [comment]
Sorry, I meant `kit.vite.server.strictPort`. Yes I agree it's not beginner-friendly, however I think the best fix is in Vite itself directly, that way it'll benefit anyone using Vite too.

**Comment by bluwy** [comment]
Actually I just tested in Vite and it did print a human-friendly message. Maybe it's the way we call the Vite dev server, or we're calling the port earlier before Vite? I haven't dug deep into this but I'm curious why Vite's message doesn't surface.

File: packages/kit/src/cli.js [hunk 1]
```diff
@@ -85,10 +85,13 @@ prog
 	.option('-H, --https', 'Use self-signed HTTPS certificate')
 	.option('-o, --open', 'Open a browser tab')
 	.action(async ({ port, host, https, open }) => {
-		await check_port(port);
-
 		process.env.NODE_ENV = process.env.NODE_ENV || 'development';
 		const config = await get_config();
+		const vite_config = config.kit.vite();
+
+		if (!vite_config.server || vite_config.server.strictPort !== false) {
```
Comment (Zerotask) [inline packages/kit/src/cli.js:92]: As a work-around for the moment: if server is not set or strictPort is not false, do a check port

**Comment by Rich-Harris** [comment]
Can you describe the steps to reproduce this? This is what I see if I `npm run dev` while the port is occupied (with a freshly made project, no alterations to the config):

image

As an aside, I notice that we're calling `config.vite()` twice — once in `cli.js`, once in `dev.js`. This PR didn't introduce it, but if we end up changing things around then it's probably a good opportunity to clean it up. Maybe we should attach the relevant info to the `watcher`, and only call `config.vite()` inside `dev.js`?

**Comment by Zerotask** [comment]
I just tested it again and still got the error:
> listen EACCES: permission denied 127.0.0.1:3000
Error: listen EACCES: permission denied 127.0.0.1:3000
    at Server.setupListenHandle [as _listen2] (node:net:1317:21)
    at listenInCluster (node:net:1382:12)
    at doListen (node:net:1520:7)
    at processTicksAndRejections (node:internal/process/task_queues:84:21)

I followed the steps from kit.svelte.dev:
> npm init svelte@next my-app
cd my-app
npm install
npm run dev -- --open

I did not change any config files. I just created the svelte demo app and played around with some svelte files, nothing special.

**Comment by Rich-Harris** [comment]
Looking into this further, it seems that there's an underlying Vite issue: https://github.com/vitejs/vite/issues/5801. It's probably best that we don't add temporary code to work around that issue, so I'll close this. Thank you

**Pull request closed without merging**